rom/dev/src/flow/cold_reset/crypto.rs

```diff
@@ -136,17 +137,26 @@ impl Crypto {
         seed: KeyId,
         priv_key: KeyId,
     ) -> CaliptraResult<Ecc384KeyPair> {
-        // [TODO] Add Nonce to the ecc384_key_gen function
         let seed = Ecc384Seed::Key(KeyReadArgs::new(seed));
 
         let mut usage = KeyUsage::default();
         usage.set_ecc_private_key(true);
 
         let key_out = Ecc384PrivKeyOut::Key(KeyWriteArgs::new(priv_key, usage));
 
+        // Generate a nonce.
+        let mut nonce: [u32; 12] = [0u32; 12];
+        let num_words = NonZeroUsize::new(nonce.len()).unwrap();
+        let _ = env.csrng.generate(num_words)?;
+        for (dst, src) in zip(&mut nonce, env.csrng.generate(num_words)?) {
+            *dst = src;
+        }
+
         Ok(Ecc384KeyPair {
             priv_key,
-            pub_key: env.ecc384.key_pair(seed, &Array4x12::default(), key_out)?,
+            pub_key: env
+                .ecc384
+                .key_pair(seed, &Array4x12::from(nonce), key_out)?,
```

Comment:
Won't adding this nonce break the certificates issued for IDevID and AliasFMC? The CSRNG would be seeded differently on each boot, yeah?

Reply:
Yeah, I don't think we want a random nonce. Perhaps you mean to feed the entropy to the [IV register](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.ecc_reg.ECC_IV%5B0%5D)

From the [hw spec](https://github.com/chipsalliance/caliptra-rtl/blob/main/docs/Caliptra_Hardware_Specification.pdf):

> Implementation of complete unified addition formula for the scalar multiplication avoids
information leakage and enhances architecture from security/mathematical perspective.
To protect the architecture against horizontal power/EM and differential power analysis (DPA)
attacks, two countermeasures are embedded to the design [9]. Since these countermeasures
require random inputs, HMAC-DRBG is fed by “IV” to generate these random values.
Since HMAC-DRBG generates random value in a deterministic way, firmware MUST feed
different IV to ECC engine for EACH keygen and signing operations.
The prior work shows that the countermeasure still shows enough resistance even though the
attack somehow finds a stronger attack mechanism.

Reply:
Correct, this should have been fed to the IV register. Updated.